File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let obj={}
    let majority = Math.floor(nums.length/2)
    for(let num of nums){
        obj[num] = (obj[num] || 0) +1;
        if(obj[num] > majority){
            
            return num;
        }
    }
}

File: 169-majority-element/majority-element.cs
public class Solution {
     public int MajorityElement(int[] nums)
    {
        Dictionary<int, int> dict = new Dictionary<int,int>();
        for(int i=0;i<nums.Length; i++)
        {
            if(dict.ContainsKey(nums[i])){
                
                dict[nums[i]]++;
                
            }
            else{
            dict[nums[i]] = 1;
            }
            if(dict[nums[i]] >(nums.Length)/2) return nums[i];
        }
       return -1; 
    }

}